### PR TITLE
Update the miner to use the new hugging face and chain stores

### PR DESCRIPTION
**Pull request by RusticLuftig**
NOTE: This change hasn't yet been fully tested. Pushing now to reduce future merge conflicts.

1. Updates the miner to push models to HF
2. Minor updates to the tools/ directory (mostly due to constant rename) that also caused a formatting change
3. Minor clean-up of some validator stuff
4. Moves the PerfMonitor into utilities/ and delete utils/

### CODE REVIEW

File: model/storage/hugging_face/hugging_face_model_store.py [hunk 1]
```diff
@@ -12,6 +12,12 @@
 class HuggingFaceModelStore(RemoteModelStore):
     """Hugging Face based implementation for storing and retrieving a model."""
 
+    @classmethod
+    def assert_access_token_exists(cls):
```
Comment (Sid-Data-Universe) [inline model/storage/hugging_face/hugging_face_model_store.py:16]: Did you want to replace the check in upload_model with this?

File: utilities/utils.py [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
+
+    return parts[0], parts[1]
+
+
+def run_in_subprocess(func: functools.partial, ttl: int) -> Any:
+    """Runs the provided function on a subprocess with 'ttl' seconds to complete.
 
     Args:
-        func (functools.partial): Function to be run.
+        func (functools.partial): Function to be run. MUST NOT BE A NESTED FUNCTION
```
Comment (Sid-Data-Universe) [inline utilities/utils.py:36]: Can remove this extra comment now right?

File: model/storage/chain/constants.py [hunk 3]
```diff
@@ -1,2 +0,0 @@
-# The uid for this subnet.
-SUBNET_UID = 9
```
Comment (Sid-Data-Universe) [inline model/storage/chain/constants.py:2]: Does the chain metadata store not need to be updated to get the subnet uid from pretrain now?

In retrospect maybe this should just be at a higher level in the repo?

File: neurons/miner.py [hunk 4]
```diff
@@ -129,143 +137,238 @@ def get_config():
...
+    actions: Actions, config: bt.config, metagraph: bt.metagraph, model_dir: str
+) -> PreTrainedModel:
+    """Loads the model to train based on the provided config."""
+
+    # Initialize the model based on the best on the network.
+    if config.load_best:
+        # Get the best UID be incentive and load it.
+        best_uid = pt.graph.best_uid(metagraph)
+        model = await actions.load_remote_model(best_uid, metagraph, model_dir)
+        bt.logging.success(f"Training with best uid: {best_uid}")
```
Comment (Sid-Data-Universe) [inline neurons/miner.py:168]: nit: "with model from best uid" to align with log below.

File: pretrain/mining.py [hunk 5]
```diff
@@ -17,389 +17,97 @@
...
-    """
-    return os.path.expanduser(
-        "{}/{}/{}/netuid{}/{}".format(
-            bt.logging.config().logging.logging_dir,
-            wallet.name,
-            wallet.hotkey_str,
-            pt.NETUID,
-            "miner/model.safe",
+        # Save the model state to the specified path.
+        # TODO: Should this be a filepath or a directory?
```
Comment (Sid-Data-Universe) [inline pretrain/mining.py:56]: Remove todo now that it is known to be a dir?

File: neurons/miner.py [hunk 6]
```diff
@@ -129,143 +137,238 @@ def get_config():
...
+                miner_actions.save(model, model_path)
+
+                # Also upload this better model to wandb, if enabled.
+                if use_wandb:
+                    wandb_run.save(glob_str=model_path)
+
+        bt.logging.success("Finished training")
+        # Push the model to your run.
+        if not config.offline:
+            if best_avg_loss < config.avg_loss_upload_threshold:
```
Comment (Sid-Data-Universe) [inline neurons/miner.py:346]: Should we adjust this by the pt.timestamp_epsilon as well? Or is that implied in the config value?

File: model/storage/chain/constants.py (hunk 3 above)
Comment (RusticLuftig) [reply model/storage/chain/constants.py:2]: Already done

File: neurons/miner.py (hunk 6 above)
Comment (RusticLuftig) [reply neurons/miner.py:346]: This is simply the threshold that the user specifies. I don't think we need to adjust.

**Review by Sid-Data-Universe — APPROVED** [review]

**Pull request merged**